plugins/rapture/nexus-coreui-plugin/src/main/java/org/sonatype/nexus/coreui/ComponentComponent.groovy

```diff
@@ -53,35 +57,160 @@ class ComponentComponent
+      def queryParams = null
+      def filter = parameters.getFilter('filter')
+      if (filter) {
+        whereClause = "${StorageFacet.P_NAME} LIKE :nameFilter OR ${StorageFacet.P_GROUP} LIKE :groupFilter OR ${StorageFacet.P_VERSION} LIKE :versionFilter"
+        queryParams = [
+            'nameFilter': "%${filter}%",
+            'groupFilter': "%${filter}%",
+            'versionFilter': "%${filter}%"
+        ]
+      }
+
+      List<ComponentXO> results = storageTx.findComponents(whereClause, queryParams, repositories, querySuffix)
+          .collect { component ->
+        new ComponentXO(
+            id: component.entityMetadata.id,
+            repositoryName: repository.name,
+            group: component.group(),
+            name: component.name(),
+            version: component.version(),
+            format: component.format()
+        )
+      }
+      if (parameters.start == 0 && !filter && readUnattachedAssets(storageTx, repositories).size()) {
+        results = new ArrayList<>(results)
+        results.add(0, new ComponentXO(
+            id: UNATTACHED,
+            repositoryName: repository.name
+        ))
+      }
+      return new PagedResponse<ComponentXO>(
+          (results.size() < parameters.limit ? 0 : parameters.limit) + results.size() + parameters.start,
+          results
+      )
+    }
+    finally {
+      storageTx.close()
+    }
   }
 
-  @Validate
-  List<AssetXO> readAssets(final @NotNull String repositoryName, final @NotNull String componentId) {
-    Repository repository = repositoryManager.get(repositoryName)
+  @DirectMethod
+  List<AssetXO> readAssets(final StoreLoadParameters parameters) {
+    Repository repository = repositoryManager.get(parameters.getFilter('repositoryName'))
     securityHelper.ensurePermitted(new RepositoryViewPermission(repository, READ))
 
     if (!repository.configuration.online) {
       return null
     }
+
+    def componentId = parameters.getFilter('componentId')
+
     StorageTx storageTx = repository.facet(StorageFacet).openTx()
     try {
-      Component component = storageTx.findComponent(new EntityId(componentId), storageTx.getBucket())
-      if (component == null) {
-        return null
+      def repositories
+      try {
+        repositories = repository.facet(GroupFacet).leafMembers()
+      }
+      catch (MissingFacetException e) {
+        repositories = ImmutableList.of(repository)
       }
 
-      return storageTx.browseAssets(component).collect {asset ->
-        new AssetXO(
-            id: asset.entityMetadata.id,
-            name: asset.name() ?: component.name(),
-            contentType: asset.contentType()
-        )
+      if (componentId == UNATTACHED) {
+        return readUnattachedAssets(storageTx, repositories)
+      }
+      else if (repositories.size() == 1) {
+        Component component = storageTx.findComponent(new EntityId(componentId), storageTx.getBucket())
+        if (component == null) {
```

Comment:
This looks like an error condition to me - we sent this componentId to the UI but now we can't find it? Might be subject to mutation of the underlying data between calls, but probably worth at least logging...